File: Programmers/Lv.0/조건에 맞게 수열 변환하기 2/181881.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        int answer = -1;
        
        int[] arr2 = new int[arr.length];
        boolean flag = true;
        
        while (flag) {
            answer++;
            arr2 = Arrays.copyOf(arr, arr.length);
            
            for (int i=0; i<arr.length; i++) {
                if (arr[i] >= 50 && arr[i] % 2 == 0) {
                    arr[i] /= 2;
                } else if (arr[i] < 50 && arr[i] % 2 != 0) {
                    arr[i] *= 2;
                    arr[i]++;
                }
            }
            
            for (int j=0; j<arr.length; j++) {
                if (arr[j] != arr2[j]) {
                    flag = true;
                    break;
                } else {
                    flag = false;
                }
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.0/조건에 맞게 수열 변환하기 2/181881.js
function solution(arr) {
    let answer = -1;
    
    let arr2 = [];
    while (!arr.every((v, i) => v === arr2[i])) {
        answer++;
        arr2 = [...arr];
        
        for (let i=0; i<arr.length; i++) {
            if (arr[i] >= 50 && arr[i] % 2 === 0) {
                arr[i] /= 2;
            } else if (arr[i] < 50 && arr[i] % 2 !== 0) {
                arr[i] *= 2;
                arr[i]++;
            }
        }
    }
    
    return answer;
}
